python/pyiceberg/catalog/rest.py

```diff
@@ -0,0 +1,373 @@
+
+    def _handle_non_200_response(self, exc: HTTPError, error_handler: Dict[int, Type[Exception]]):
+        try:
+            response = ErrorResponse(**exc.response.json())
+        except Exception:  # pylint: disable=W0012,W0703
+            # In the case we don't have a proper response
+            response = ErrorResponse(
+                error=ErrorResponseMessage(
+                    message=f"Could not decode json payload: {exc.response.text}",
+                    type="RESTError",
+                    code=exc.response.status_code,
+                )
+            )
+
+        code = exc.response.status_code
+        if code in error_handler:
+            raise error_handler[code](response) from exc
+        elif code == 400:
+            raise BadRequestError(response.error.message) from exc
+        elif code == 401:
+            raise UnauthorizedError(response.error.message) from exc
+        elif code == 503:
+            raise ServiceUnavailableError(response.error.message) from exc
+        elif 500 <= code < 600:
+            raise ServerError(response.error.message) from exc
+        else:
+            raise RESTError(response.error.message) from exc
+
+    def create_table(
+        self,
+        identifier: Union[str, Identifier],
+        schema: Schema,
+        location: Optional[str] = None,
+        partition_spec: PartitionSpec = UNPARTITIONED_PARTITION_SPEC,
+        sort_order: SortOrder = UNSORTED_SORT_ORDER,
+        properties: Optional[Properties] = None,
+    ) -> RestTable:
+        namespace_and_table = self._split_namespace_and_table(identifier)
+        properties = properties or {}
+        request = CreateTableRequest(
+            name=namespace_and_table["table"],
+            location=location,
+            table_schema=schema,
+            partition_spec=partition_spec,
+            write_order=sort_order,
+            properties=properties,
+        )
+        serialized_json = request.json()
+        response = requests.post(
+            self.url(Endpoints.create_table, namespace=namespace_and_table["namespace"]),
+            data=serialized_json,
+            headers=self.headers,
+        )
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {409: TableAlreadyExistsError})
+
+        return RestTable(identifier=identifier, **response.json())
+
+    def list_tables(self, namespace: Optional[Union[str, Identifier]] = None) -> List[Identifier]:
+        namespace_concat = NAMESPACE_SEPARATOR.join(self.identifier_to_tuple(namespace or ""))
+        response = requests.get(
+            self.url(Endpoints.list_tables, namespace=namespace_concat),
+            headers=self.headers,
+        )
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {404: NoSuchTableError})
```

Comment:
404 here should be `NoSuchNamespaceError` because there isn't a specific table.

Reply:
Got it.